internal/api/phone.go

```diff
@@ -95,7 +95,8 @@ func (a *API) sendPhoneConfirmation(ctx context.Context, r *http.Request, tx *st
 		if err != nil {
 			return "", err
 		}
-		if config.Hook.SendSMS.Enabled {
+		// We call sendPhoneConfirmation even when autoconfirm is enabled
+		if config.Hook.SendSMS.Enabled && !config.Sms.Autoconfirm {
```

Comment:
nit: reorder for readability (easier to notice the !)
```suggestion
		// Hook should only be called if SMS autoconfirm is disabled
		if !config.Sms.Autoconfirm && config.Hook.SendSMS.Enabled {
```